### PR TITLE
feat: add snaplet command completion spec

### PR DESCRIPTION
**Pull request by avallete**
Hey there !

Snaplet is a tool to help user deal with their database: https://www.snaplet.dev/

I love Fig so I though I want this tool to have those superpower as well !

### CODE REVIEW

File: src/snaplet.ts [hunk 1]
```diff
@@ -0,0 +1,608 @@
...
+};
+const completionCommand: Fig.Subcommand = {
+  options: [...commonCommandsOptions],
+  name: "completion",
+  description: "Generate completion script",
+};
+
+const completionSpec: Fig.Spec = {
+  name: "snaplet",
+  description: "",
```
Comment (fedeci) [inline src/snaplet.ts:588]: Can you please provide a description for the tool?

File: src/snaplet.ts [hunk 2]
```diff
@@ -0,0 +1,608 @@
...
+        description: `${database.size} (${database.snapshotName})`,
+      });
+    }
+    return result;
+  },
+};
+
+const commonCommandsOptions = [
+  {
+    name: ["--help", "-h"],
```
Comment (fedeci) [inline src/snaplet.ts:121]: With this property the option becomes automatically available to all subcommands. In such way you don't have to worry about adding `options: [...commonCommandsOptions]`. Can you please remove those?
```suggestion
    name: ["--help", "-h"],
    isPersistent: true,
```

File: src/snaplet.ts [hunk 3]
```diff
@@ -0,0 +1,608 @@
...
+    description: "Show help for snaplet",
+  },
+  {
+    name: ["--version", "-v"],
+    description: "Show version number",
```
Comment (fedeci) [inline src/snaplet.ts:126]:
```suggestion
    description: "Show version number",
    isPersistent: true,
```

**Review by fedeci — CHANGES_REQUESTED** [review]
Thanks @avallete! I left a couple notes to fix before merging, sorry for the late review!

File: src/snaplet.ts (hunk 2 above)
Comment (avallete) [reply src/snaplet.ts:121]: Oh nice ! I was sure they were a better way to do it. Thank's for the suggestion !

File: src/snaplet.ts (hunk 1 above)
Comment (avallete) [reply src/snaplet.ts]: Done 👍

**Comment by avallete** [comment]
> Thanks @avallete! I left a couple notes to fix before merging, sorry for the late review!

Hey there! No worries, I see the number of PRs on the repo 😂. I can imagine how busy things must be, so I truly appreciate you taking the time to review my contribution and for the insightful suggestions you provided!

I've made the necessary changes based on your feedback.

**Review by fedeci — APPROVED** [review]
Thanks!

File: src/snaplet.ts [hunk 4]
```diff
@@ -0,0 +1,616 @@
...
+          description: "Path to a sample config file",
+          args: {
+            template: "filepaths",
+          },
+        },
+        {
+          name: ["-f", "--force"],
+          description: "Force overwrite existing sample file",
+          isDangerous: true,
+          isRequired: false,
```
Comment (fedeci) [inline src/snaplet.ts:554]:
```suggestion
```

**Pull request merged**